core-bundle/src/Controller/FrontendModule/FeedReaderController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller\FrontendModule;
+
+use Contao\Config;
+use Contao\CoreBundle\Exception\PageNotFoundException;
+use Contao\CoreBundle\ServiceAnnotation\FrontendModule;
+use Contao\CoreBundle\Twig\FragmentTemplate;
+use Contao\Environment;
+use Contao\ModuleModel;
+use Contao\Pagination;
+use Contao\StringUtil;
+use FeedIo\Feed;
+use FeedIo\Feed\Item;
+use FeedIo\FeedInterface;
+use FeedIo\FeedIo;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+/**
+ * @FrontendModule(category="miscellaneous")
+ */
+class FeedReaderController extends AbstractFrontendModuleController
+{
+    public function __construct(private readonly FeedIo $feedIo, private readonly LoggerInterface $logger, private readonly CacheInterface $cache)
+    {
+    }
+
+    protected function getResponse(FragmentTemplate $template, ModuleModel $model, Request $request): Response
+    {
+        $this->initializeContaoFramework();
+
+        $feeds = [];
+
+        foreach (StringUtil::trimsplit('[\n\t ]', trim($model->feed_urls)) as $url) {
+            try {
+                $feed = $this->cache->get('feed_reader_'.$model->id.'_'.md5($url), function (ItemInterface $item) use ($url, $model) {
```

Comment:
Why do we access the system cache here? Isn't it enough to cache the whole response with the HTTP Cache if needed?

Reply:
that's how the RSS feed module works. It has a configuration for how long to cache the feeds, which is independent from how long the page is cached. I would refrain from discussing ESI options or thelike at the current stage of development (and Contao 5 release time).

Reply:
Also, I keep saying this in all kinds of PRs and extensions: Just because we have an HTTP cache does not mean you shouldn't use any other cache anymore. The HTTP cache is just another layer on top and it can be disabled if your logged in (as FE or BE user) or working in DEV mode or preview mode etc. All these cases would cause live requests which are completely unnessecary. So this is definitely correct here.